File: objc/Project14/Project14/GameView.m
//
//  GameView.m
//  Project14
//
//  Created by Jinwoo Kim on 1/12/21.
//

#import "GameView.h"

@implementation GameView

- (void)resetCursorRects {
    [super resetCursorRects];
    
    NSImage *targetImage = [NSImage imageNamed:@"cursor"];
    if (targetImage) {
        NSCursor *cursor = [[NSCursor alloc] initWithImage:targetImage
                                                   hotSpot:NSMakePoint(targetImage.size.width / 2, targetImage.size.height / 2)];
        
        [self addCursorRect:self.frame cursor:cursor];
    }
}

@end

File: swift/Project14/Project14/GameView.swift
//
//  GameView.swift
//  Project14
//
//  Created by Jinwoo Kim on 1/12/21.
//

import SpriteKit

class GameView: SKView {
    override func resetCursorRects() {
        super.resetCursorRects()
        
        if let targetImage = NSImage(named: "cursor") {
            let cursor = NSCursor(image: targetImage, hotSpot: CGPoint(x: targetImage.size.width / 2, y: targetImage.size.height / 2))
            addCursorRect(frame, cursor: cursor)
        }
    }
}
